Source/Scene/ShadowMap.js

```diff
@@ -307,110 +448,308 @@ define([
+
+        for (var j = 0; j < numberOfCascades; ++j) {
+            // Find the bounding box of the camera sub-frustum in shadow map texture space
+            cascadeSubFrustum.near = splitsNear[j];
+            cascadeSubFrustum.far = splitsFar[j];
+            var viewProjection = Matrix4.multiply(cascadeSubFrustum.projectionMatrix, sceneCamera.viewMatrix, scratchMatrix);
+            var inverseViewProjection = Matrix4.inverse(viewProjection, scratchMatrix);
+            var cascadeMatrix = Matrix4.multiply(shadowMapCamera.getViewProjection(), inverseViewProjection, scratchMatrix);
+
+            // Project each corner from camera NDC space to shadow map texture space. Min and max will be from 0 to 1.
+            var min = Cartesian3.fromElements(Number.MAX_VALUE, Number.MAX_VALUE, Number.MAX_VALUE, scratchMin);
+            var max = Cartesian3.fromElements(-Number.MAX_VALUE, -Number.MAX_VALUE, -Number.MAX_VALUE, scratchMax);
+
+            for (var k = 0; k < 8; ++k) {
+                var corner = Cartesian4.clone(frustumCornersNDC[k], scratchFrustumCorners[k]);
+                Matrix4.multiplyByVector(cascadeMatrix, corner, corner);
+                Cartesian3.divideByScalar(corner, corner.w, corner); // Handle the perspective divide
+                Cartesian3.minimumByComponent(corner, min, min);
+                Cartesian3.maximumByComponent(corner, max, max);
+            }
+
+            // Limit min to 0.0, sometimes precision errors cause it to go slightly negative.
+            min.x = Math.max(min.x, 0.0);
+            min.y = Math.max(min.y, 0.0);
+            min.z = 0.0; // Always start cascade frustum at the top of the light frustum to capture objects in the light's path
+
+            var cascadeCamera = shadowMap._cascadeCameras[j];
+            cascadeCamera.clone(shadowMapCamera); // PERFORMANCE_IDEA : could do a shallow clone for all properties except the frustum
+            cascadeCamera.frustum.left = left + min.x * (right - left);
+            cascadeCamera.frustum.right = left + max.x * (right - left);
+            cascadeCamera.frustum.bottom = bottom + min.y * (top - bottom);
+            cascadeCamera.frustum.top = bottom + max.y * (top - bottom);
+            cascadeCamera.frustum.near = near + min.z * (far - near);
+            cascadeCamera.frustum.far = near + max.z * (far - near);
+
+            // Scale and offset are used to transform shadowPosition to the proper cascade in the shader
+            var cascadeScale = shadowMap._cascadeScales[j];
+            cascadeScale.x = 1.0 / (max.x - min.x);
+            cascadeScale.y = 1.0 / (max.y - min.y);
+            cascadeScale.z = 1.0 / (max.z - min.z);
+
+            var cascadeOffset = shadowMap._cascadeOffsets[j];
+            cascadeOffset.x = -min.x;
+            cascadeOffset.y = -min.y;
+            cascadeOffset.z = -min.z;
+        }
+    }
+
+    function createViewMatrix(d, u, r, result) {
```

Comment:
That sounds good to be, this code is nearly identical to the code in `Camera.js`